web/src/compose_actions.ts

```diff
@@ -179,7 +210,21 @@ function same_recipient_as_before(opts) {
     );
 }
 
-export function start(opts) {
+type ComposeActionsStartOpts = {
+    message_type: "private" | "stream";
+    force_close?: boolean;
+    trigger?: string;
```

Comment:
Does it make sense for `trigger` to be required? I don't know to what extent we have many code paths without it, but I think the original design intent had been for it to be mandatory.

Reply:
Probably a good idea as a follow-up, but not really essential; it's purpose is 90% for logging/Sentry, 10% for the bits of code that behave differently off it, which would largely be better reworked.